src/test/java/com/google/devtools/build/lib/unsafe/StringUnsafeTest.java

```diff
@@ -44,8 +44,13 @@ public void testGetBytes() {
     StringUnsafe stringUnsafe = StringUnsafe.getInstance();
     assertThat(ByteBuffer.wrap(stringUnsafe.getByteArray("hello")))
         .isEqualTo(StandardCharsets.ISO_8859_1.encode("hello"));
-    assertThat(ByteBuffer.wrap(stringUnsafe.getByteArray("lambda λ")))
-        .isEqualTo(StandardCharsets.UTF_16LE.encode("lambda λ"));
+
+    if (System.getProperty("os.arch").equals("s390x"))
```

Comment:
Instead of hardcoding the s390x arch here, could we instead use e.g.

```java
import java.nio.ByteOrder;

if (ByteOrder.nativeOrder().equals(ByteOrder.BIG_ENDIAN)) {
  ...
}
```

?

Reply:
Thanks for reviewing. Yes, I will try this change and upload a new patch once I have results on s390x.

Reply:
Updated this PR and commit message, this test still passes on s390x.